### PR TITLE
[4] Display arial-label only if different from button text

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue # .

### Summary of Changes
This pr checks if the text on the readme button is visible in ful length. If yes, the aria-label is redundant and not displayed.

### Testing Instructions
Test wit different settings for readmore buttons: With and without titile, with own text, with and without a fixed number for the length. 
Inspect then a readmore-button. 

### Actual result BEFORE applying this Pull Request
If the text on the readmore button shows the full title of the article, aria-label and button text are the same

grafik

### Expected result AFTER applying this Pull Request
If the text on the readmore button shows the full title of the article, there is no aria-label

### CODE REVIEW

**Comment by chmst** [comment]
Fixed string for comparision, thanks @christineWK

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on cf197e6e73457547af93e68902e8edef2236538f<br /><br />Now it's OK :-)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35491">issues.joomla.org/tracker/joomla-cms/35491</a>.</sub>

File: layouts/joomla/content/readmore.php [hunk 1]
```diff
@@ -38,8 +38,10 @@
 			<?php echo Text::_('JGLOBAL_READ_MORE'); ?>
 		</a>
 	<?php else : ?>
-		<a class="btn btn-secondary" href="<?php echo $displayData['link']; ?>" aria-label="<?php echo Text::sprintf('JGLOBAL_READ_MORE_TITLE', $this->escape($item->title)); ?>">
-			<?php echo '<span class="icon-chevron-' . $direction . '" aria-hidden="true"></span>'; ?>
+		<?php $text = Text::sprintf('JGLOBAL_READ_MORE_TITLE', HTMLHelper::_('string.truncate', $item->title, $params->get('readmore_limit'))); ?>
+		<?php $ariaLabel = Text::sprintf('JGLOBAL_READ_MORE_TITLE', $this->escape($item->title)); ?>
+		<a class="btn btn-secondary" href="<?php echo $displayData['link']; ?>" <?php echo $text != $ariaLabel ? 'aria-label="' . $ariaLabel . '"' : ''; ?>>
+			<?php echo '<span class="icon-chevron-' . $direction . '" aria-hidden="true"></span>';?>
 			<?php echo Text::sprintf('JGLOBAL_READ_MORE_TITLE', HTMLHelper::_('string.truncate', $item->title, $params->get('readmore_limit'))); ?>
```
Comment (Quy) [inline layouts/joomla/content/readmore.php:45]:
```suggestion
			<?php echo $text; ?>
```

Comment (Quy) [inline layouts/joomla/content/readmore.php]:
```suggestion
			<?php echo '<span class="icon-chevron-' . $direction . '" aria-hidden="true"></span>'; ?>
			<?php echo $text; ?>
```

**Comment by brianteeman** [comment]
This change is not really needed. It doesn't solve a problem

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on da91076d15a0458f1502c1399b538bcaaa5227a2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35491">issues.joomla.org/tracker/joomla-cms/35491</a>.</sub>

File: layouts/joomla/content/readmore.php [hunk 2]
```diff
@@ -38,9 +38,11 @@
 			<?php echo Text::_('JGLOBAL_READ_MORE'); ?>
 		</a>
 	<?php else : ?>
-		<a class="btn btn-secondary" href="<?php echo $displayData['link']; ?>" aria-label="<?php echo Text::sprintf('JGLOBAL_READ_MORE_TITLE', $this->escape($item->title)); ?>">
+		<?php $text = Text::sprintf('JGLOBAL_READ_MORE_TITLE', HTMLHelper::_('string.truncate', $item->title, $params->get('readmore_limit'))); ?>
+		<?php $ariaLabel = Text::sprintf('JGLOBAL_READ_MORE_TITLE', $this->escape($item->title)); ?>
+		<a class="btn btn-secondary" href="<?php echo $displayData['link']; ?>" <?php echo $text != $ariaLabel ? 'aria-label="' . $ariaLabel . '"' : ''; ?>>
 			<?php echo '<span class="icon-chevron-' . $direction . '" aria-hidden="true"></span>'; ?>
-			<?php echo Text::sprintf('JGLOBAL_READ_MORE_TITLE', HTMLHelper::_('string.truncate', $item->title, $params->get('readmore_limit'))); ?>
+			<?php echo $text; ?>
```
Comment (PhilETaylor) [inline layouts/joomla/content/readmore.php]: Surely this needs escaping if its user provided text?

**Comment by khu5h1** [comment]
I have tested this item :white_check_mark: successfully on da91076d15a0458f1502c1399b538bcaaa5227a2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35491">issues.joomla.org/tracker/joomla-cms/35491</a>.</sub>

**Comment by richard67** [comment]
@chmst Please check the review comment by Phil. I think he is right and the title needs to be escaped `$this->escape($item->title)` in line 41.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35491">issues.joomla.org/tracker/joomla-cms/35491</a>.</sub>

File: layouts/joomla/content/readmore.php [hunk 3]
```diff
@@ -38,9 +38,11 @@
 			<?php echo Text::_('JGLOBAL_READ_MORE'); ?>
 		</a>
 	<?php else : ?>
-		<a class="btn btn-secondary" href="<?php echo $displayData['link']; ?>" aria-label="<?php echo Text::sprintf('JGLOBAL_READ_MORE_TITLE', $this->escape($item->title)); ?>">
+		<?php $text = Text::sprintf('JGLOBAL_READ_MORE_TITLE', HTMLHelper::_('string.truncate', $item->title, $params->get('readmore_limit'))); ?>
```
Comment (HLeithner) [inline layouts/joomla/content/readmore.php]:
```suggestion
		<?php $text = Text::sprintf('JGLOBAL_READ_MORE_TITLE', $this->escape(HTMLHelper::_('string.truncate', $item->title, $params->get('readmore_limit')))); ?>
```
As suggested by Phil

File: layouts/joomla/content/readmore.php [hunk 4]
```diff
@@ -38,9 +38,11 @@
...
+		<?php $ariaLabel = Text::sprintf('JGLOBAL_READ_MORE_TITLE', $this->escape($item->title)); ?>
+		<a class="btn btn-secondary" href="<?php echo $displayData['link']; ?>" <?php echo $text != $ariaLabel ? 'aria-label="' . $ariaLabel . '"' : ''; ?>>
```
Comment (HLeithner) [inline layouts/joomla/content/readmore.php]:
```suggestion
		<a class="btn btn-secondary" href="<?php echo $displayData['link']; ?>" <?php echo $text !== $ariaLabel ? 'aria-label="' . $ariaLabel . '"' : ''; ?>>
```

File: layouts/joomla/content/readmore.php (hunk 2 above)
Comment (chmst) [reply layouts/joomla/content/readmore.php]: thx

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on b6b12aab156467f3bb1f6ebb8e6abe11e375dda0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35491">issues.joomla.org/tracker/joomla-cms/35491</a>.</sub>

**Comment by richard67** [comment]
From "needs review" back to "pending" in the issue tracker.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35491">issues.joomla.org/tracker/joomla-cms/35491</a>.</sub>

**Comment by Quiviro** [comment]
I have tested this item :red_circle: unsuccessfully on ded6b69ebbcbee5affea03bebc30f8ca49de1012<br /><br />


If I set a custom text for the button, the aria label still exists. With default settings for "view more" button, the patch does work.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35491">issues.joomla.org/tracker/joomla-cms/35491</a>.</sub>

**Comment by HLeithner** [comment]
This pull request has automatically rebased to 4.2-dev.

**Pull request closed without merging**